### PR TITLE
feat(masthead): Add a logic that highlights the current and parent menu items within the Masthead dropdown + Mega Menu based on the URL

### PR DESCRIPTION
**Pull request by huntermacd**
### Related Ticket(s)

https://jsw.ibm.com/browse/HC-2132
#6638 

### Description

This work adds highlighting/selected state to the menu item that matches the page the user is currently viewing. In desktop, the parent menu item is selected while in mobile, the selected menu item receives the styling. This way the user can better understand where they are relative to the rest of the content on the site.

Screen Shot 2021-08-20 at 10 27 55 AM

### Steps to Test
1. Go to the CodeSandbox example [here](https://codesandbox.io/s/github/annawen1/wc-masthead-selected-logic-csb?file=/index.html)
2. Click on the menu items in the iframe window

### Changelog

**New**

- n/a

**Changed**

- adds hasChildLink function to masthead-composite that determines top-level link highlight
- updates `renderNavItems` and `renderLeftNavMenuSections` functions

**Removed**

- n/a

### CODE REVIEW

**Comment by annawen1** [comment]
@huntermacd we actually want to keep the `selected-menu-item` as an option for adopters where they might run into issues with the url and may prefer setting it manually.

File: packages/web-components/src/components/masthead/masthead-composite.ts [hunk 1]
```diff
@@ -396,6 +434,8 @@ class DDSMastheadComposite extends LitElement {
     target: NAV_ITEMS_RENDER_TARGET;
     hasL1: boolean;
   }) {
+    // const currentUrlPath= window.location.href;
+    const currentUrlPath = 'https://www.ibm.com/cloud/ai';
```
Comment (kennylam) [inline packages/web-components/src/components/masthead/masthead-composite.ts:438]: The commented line should be used here, no?

Comment (huntermacd) [reply packages/web-components/src/components/masthead/masthead-composite.ts]: Yessir. I will uncomment that line when I'm ready to pull the PR out of draft status. Thank you for the review.

**Comment by proeung** [comment]
@huntermacd Glad to see that this PR is out of the draft status! Also, looks like there are some conflicting files, can you help resolve this?

**Comment by shixiedesign** [comment]
Happy to look but needing some testing instruction here. I know we agreed on highlighting the first matching parent to get the Selected styling (blue underline), but not sure how I can QA that visually.

**Comment by huntermacd** [comment]
Publishing this PR with a couple caveats:

- testing will be tricky because the code is relying on `window.location.href`, which is hard to mock in Storybook
- the work covers most cases with the exception of the Cloud megamenu tabs, which will need to be handled in a follow-up

**Comment by annawen1** [comment]
@huntermacd I've set up a codesandbox that should allow us to test the logic [here](https://codesandbox.io/s/github/annawen1/wc-masthead-selected-logic-csb?file=/index.html)

@shixiedesign you should be able to test with the above link (make sure to open the sandbox in an external window like I mentioned in the description of this PR)

**Comment by proeung** [comment]
@huntermacd Thanks for the update!

The remaining of this work has been reassigned to @annawen1 so that we can get everything in place and ready for testing.

**Comment by annawen1** [comment]
@huntermacd leaving a note here, after speaking with @shixiedesign I've removed the logic adding the active state to the megamenu items as the functional and design specs don't specify those changes. Design intent is that we are only adding the blue border "selected" state to the parent menu items and in the mobile nav items.

**Review by kennylam — APPROVED** [review]
LGTM!

**Review by shixiedesign — APPROVED** [review]
It’s fantastic! Thank you for getting this done so quickly! :thumbsup:

**Comment by jeffchew** [comment]
@huntermacd could you resolve the merge conflicts?

**Comment by jeffchew** [comment]
> @huntermacd could you resolve the merge conflicts?

Thank you for resolving, @annawen1 !

**Pull request merged**